src/utils_SymKLIEPLoss.jl

```diff
@@ -0,0 +1,90 @@
+####################################
+#
+# symmetric KLIEP loss
+#
+####################################
+struct CDSymKLIEPLoss <: CoordinateDifferentiableFunction
+    μx::Vector{Float64}   # mean(Ψx, dims = 2)
+    μy::Vector{Float64}   # mean(Ψy, dims = 2)
+    Ψx::Matrix{Float64}
+    Ψy::Matrix{Float64}
+    rx::Vector{Float64}    # nx dim vector --- stores -θ'Ψx(i)
+    ry::Vector{Float64}    # ny dim vector --- stores  θ'Ψy(j)
+    p::Int64
+end
+
+function CDSymKLIEPLoss(Ψx::Matrix{Float64}, Ψy::Matrix{Float64})
+    (p = size(Ψx, 1)) == size(Ψy, 1) || throw(DimensionMismatch())
+
+    CDSymKLIEPLoss(vec(mean(Ψx, dims=2)), vec(mean(Ψy, dims=2)), Ψx, Ψy, zeros(size(Ψx, 2)), zeros(size(Ψy, 2)), p)
+end
+
+CoordinateDescent.numCoordinates(f::CDSymKLIEPLoss) = f.p
+
+function CoordinateDescent.initialize!(f::CDSymKLIEPLoss, x::SparseIterate)
+    mul!(f.rx, transpose(f.Ψx),-x)
```

Comment:
This needs to create a temporary array for -x.
Better to multiply transpose(f.Ψx) with x.
Then change the sign of f.rx.